core/src/main/java/com/netflix/conductor/core/dal/DomainMapper.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.netflix.conductor.core.dal;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.netflix.conductor.common.metadata.tasks.Task;
+import com.netflix.conductor.common.run.Workflow;
+import com.netflix.conductor.common.utils.ExternalPayloadStorage.Operation;
+import com.netflix.conductor.common.utils.ExternalPayloadStorage.PayloadType;
+import com.netflix.conductor.core.utils.ExternalPayloadStorageUtils;
+import com.netflix.conductor.domain.TaskDO;
+import com.netflix.conductor.domain.TaskStatusDO;
+import com.netflix.conductor.domain.WorkflowDO;
+import com.netflix.conductor.domain.WorkflowStatusDO;
+import com.netflix.conductor.metrics.Monitors;
+
+@Component
+public class DomainMapper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DomainMapper.class);
+
+    private final ExternalPayloadStorageUtils externalPayloadStorageUtils;
+
+    public DomainMapper(ExternalPayloadStorageUtils externalPayloadStorageUtils) {
+        this.externalPayloadStorageUtils = externalPayloadStorageUtils;
+    }
+
+    /**
+     * Fetch the fully formed workflow domain object with complete payloads
+     *
+     * @param workflowDO the workflow domain object from the datastore
+     * @return the workflow domain object {@link WorkflowDO} with payloads from external storage
+     */
+    public WorkflowDO getWorkflowDO(WorkflowDO workflowDO) {
+        populateWorkflowAndTaskPayloadData(workflowDO);
+        return workflowDO;
+    }
+
+    public WorkflowDO getLeanCopy(WorkflowDO workflowDO) {
+        WorkflowDO leanWorkflowDO = workflowDO.copy();
+        externalizeWorkflowData(leanWorkflowDO);
+        return leanWorkflowDO;
+    }
+
+    public Workflow getWorkflowDTO(WorkflowDO workflowDO) {
```

Comment:
Should this just be named to be `getWorkflow` instead since the returned object don't have the DTO suffix, which of course we should have ideally, but since it cannot be done due to backward compatibility. I think we should just eliminate the appear of `DTO` to avoid confusions

Reply:
changed